unit/src/symbol-table/symbol_table_util.cpp

```diff
@@ -0,0 +1,27 @@
+/*******************************************************************\
+
+ Module: Unit test utilities
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+/// \file
+/// Utility functions relating to  the sybmol table.
+
+#include "symbol_table_util.h"
+
+/// Create a lvalue symbol of the specified type.
+/// \param name: The name of the symbol.
+/// \param type: The type of the symbol.
+/// \return The symbol that has been created.
+symbolt symbol_table_utilt::create_basic_symbol(
```

Comment:
I am happy to do this, but reasons why I didn't:

 - I thought it was a bit of an ad-hoc collection of flags that seems very useful when you want succinct tests, but not in general. 
- I don't understand all the flags in `symbolt`. I don't think this is a problem for tests, but seems more risky to introduce into the general codebase (someone might see this constructor, use it but in some unrelated part of the code, the missing flag could cause some nasty problems (not an issue with unit tests since only test isolated sections of the code).